src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -113,32 +126,82 @@ export class DomEventPlugin implements Plugin {
         return '';
     }
 
+    private getElementCSSLocator(domEvent: TargetDomEvent) {
+        if (domEvent.cssLocator) {
+            return domEvent.cssLocator;
+        }
+        return undefined;
+    }
     private addEventHandler(domEvent: TargetDomEvent): void {
         const eventType = domEvent.event;
-        const eventListener = this.getEventListener();
-        this.eventListenerMap.set(domEvent, eventListener);
 
-        if (domEvent.elementId) {
+        this.eventListenerMap.set(domEvent, this.identifierToEventListenerMap);
+        const eventListenerCSS = this.getEventListener(domEvent);
+        const eventListenerID = this.getEventListener();
+        this.identifierToEventListenerMap.set('CSS', eventListenerCSS);
+        this.identifierToEventListenerMap.set('ID', eventListenerID);
+
+        let isElementIdentifiedByLocator = false;
+
+        // first add event listener to all elements identified by the CSS locator
+        if (domEvent.cssLocator) {
+            const elementList = document.querySelectorAll(domEvent.cssLocator);
+            elementList.forEach((element: HTMLElement) => {
+                if (document.getElementById(domEvent.elementId) === element) {
+                    isElementIdentifiedByLocator = true;
+                }
+                element.addEventListener(eventType, eventListenerCSS);
+            });
+        }
+
+        // add event listener to element identified by the element ID if no CSS locator provided or
+        // the element is not one of the elements identified by the CSS locator
+        if (domEvent.elementId && !isElementIdentifiedByLocator) {
             document
                 .getElementById(domEvent.elementId)
-                ?.addEventListener(eventType, eventListener);
+                ?.addEventListener(eventType, eventListenerID);
+
+            // safe default
         } else if (domEvent.element) {
```

Comment:
**issue:** If we are adding event listeners to **all** selectors (i.e., ID, object and CSS) in a given `TargetDomEvent`, this should be an `if` statement rather than an `else if` statement.

Reply:
Refactored